File: vmfiles/L3/Money.java
public class Money {
  private int dollars;
  private int cents;

  public Money (int dollars, int cents) {
    this.dollars = dollars;
    this.cents = cents;
  }

  public void add (Money moreMoney) {
    cents = cents + moreMoney.cents;
    if (cents >= 100) {
      cents = cents - 100;
     dollars++;
    }
    dollars = dollars + moreMoney.dollars;
  }

  public String toString () {
    return "$" + dollars + "." + (cents <= 9 ? "0" : "") + cents;
  }

  public static void main (String[] args) {
    Money m1 = new Money(5, 30);
    Money m2 = new Money(6, 13);

    m1.add(m2);
    System.out.println("Total is " + m1);
  }
}

File: vmfiles/L3/money.c
#include <stdio.h>

typedef struct {
  int dollars;
  int cents;
} Money;

void add (Money *someMoney, Money *moreMoney) {
  someMoney->cents = someMoney->cents + moreMoney->cents;
  if (someMoney->cents >= 100) {
    someMoney->cents = someMoney->cents - 100;
    someMoney->dollars++;
  }
  someMoney->dollars = someMoney->dollars + moreMoney->dollars;
}

int main (int argc, char *argv[]) {
  Money m1, m2;

  m1.dollars = 5;
  m1.cents = 30;

  m2.dollars = 6;
  m2.cents = 13;

  add (&m1, &m2);
  printf ("Total is $%d.%02d\n", m1.dollars, m1.cents);

  return 0;
}
